### DISCUSSION TITLE
Allow fixed (exact-size) column widths

### DISCUSSION THREAD

**Discussion by githorse** [KevinVandy/material-react-table#650] [Ideas]
I can use the column `size` prop to adjust column sizes relative to each other, and MRT uses some secret sauce to make the table look good at a lot of different screen widths. I was surprised to find out that -- unlike in other grids I've used --  `size`, `minSize`, and `maxSize` are just suggestions, behaving more like a `flex-grow` than `width`, and that the actual rendered width is rarely equal to `size`. But it turns out that this is usually what I want, because it scales well to lots of different screen widths.

But sometimes I really *do* want a fixed-width column, and as far as I can tell I can't have one. The [actions/expand/command columns](https://github.com/KevinVandy/material-react-table/issues/564) are an obvious case; I don't want those to grow dynamically, ever. I also frequently design nested table layouts where some columns in the inner grid line up with columns in the outer grid (called [aligned grids](https://www.ag-grid.com/angular-data-grid/aligned-grids/) in AG Grid); I've done this with other grids, but it seems that it's impossible to do in MRT because I can't control the widths precisely enough. In general, the inability to precisely set a column width has been a real frustration. 

Flexible relative sizing (what we have now) and fixed widths are both useful. Both [MUI Datagrid](https://mui.com/x/react-data-grid/column-dimensions/#fluid-width) and [AG Grid](https://www.ag-grid.com/javascript-data-grid/column-sizing/#column-flex), for instance, offer both `flex` and `width` props on the columns. I don't know of any datagrids besides MRT that don't support (fixed) `width`.

How about adding a mechanism (perhaps migrating to the `flex` and `width` pattern) that allows exact fixed widths?

**Comment by Bodad** [comment]
I also have a need to have very small (only a single letter) columns and can't see how to do that. 

I'm attaching a screen shot of a portion of a table.  (Rest of the table omitted for privacy issues).  But hopefully you get the gist.

![Small Data Grid](https://github.com/KevinVandy/material-react-table/assets/6472059/7525c4c9-6405-491d-8ede-5d685eba6b26)

Any help or advice from anyone would be greatly appreciated.

**Reply by KevinVandy** [reply]
That's what the new grid-no-grow layout mode is for

**Reply by Bodad** [reply]
Fantastic. I will try that out. Thank you for the quick response. Sorry it took me so long to get back to this.

**Reply by githorse** [reply]
... but I don't think `grid-no-grow` really answers the mail here, since I need to be able to mix exact-width and flexible-width columns in the same grid (as I can with other grid libraries). We still need a column sizing option that actually just fixes the size of a single column. 

This continues to be a major frustration and a big reason I'm now shopping for alternatives to MRT.
